### PR TITLE
feat: Enable Versioning

### PR DESCRIPTION
**Pull request by HanZ1203**
## Description
Enable versioning for GCP Scanner with GitHub release and tag.

## Changes Made
Add a YAML file to modify GitHub Action workflow.

## Usage 
Step 1: commit
Step 2: assign a tag. 
`git tag v1.0.0 -m 'test for versioning'`
Step 3: push.
 `git push origin v1.0.0`
Result: a release and a tag are automatically generated.

## Related Issues
#20 
related pr: #35 

## Additional Notes
This function requires configuring the workflow permissions with 'Read and Write permissions'.

### CODE REVIEW

**Comment by mshudrak** [comment]
Please accept Google CLA

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      - '*'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Create Release
+        uses: ncipollo/release-action@v1
```
Comment (mshudrak) [inline .github/workflows/release.yml:17]: Like in PR #35, we don't want extra third-party dependencies here, especially for the release process. We would prefer to explore native Github mechanism (if any) for that.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Upload asset
+        uses: svenstaro/upload-release-action@v2
```
Comment (mshudrak) [inline .github/workflows/release.yml:22]: same here.

**Review by mshudrak — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (HanZ1203) [reply .github/workflows/release.yml:17]: Understood. Thanks a lot for the advice. I will be working on that.

File: .github/workflows/release.yml (hunk 2 above)
Comment (HanZ1203) [reply .github/workflows/release.yml:22]: Understood.

File: .github/workflows/release.yml (hunk 1 above)
Comment (HanZ1203) [reply .github/workflows/release.yml]: > Like in PR #35, we don't want extra third-party dependencies here, especially for the release process. We would prefer to explore native Github mechanism (if any) for that.

I have removed the third-party dependencies and committed a new file. It works fine in my forked repository. I added the usage and notes. Please take a look:).

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      # Creat a release and upload assets.
+      # Note that the release automatically contains source code, the upload assets can be a description file of this version, e.g. change.txt
+      - name: Create Release
+        run: |
+          gh release create ${{ github.ref }} ./*.txt
```
[the remark this answers is not in the document]
Comment (HanZ1203) [reply .github/workflows/release.yml:22]: > Thanks, but we would prefer to move forward with solution in #100

Thanks for your reply. I viewed the solution in #100. The release is achieved via a GitHub action release-please-action[https://github.com/marketplace/actions/release-please-action](url). It seems still a third-party dependency created by some contributors. That's why I use the _gh_ command. 
If the release-please-action is ok, I'm glad to update my YAML file and solve this versioning issue.

[the remark this answers is not in the document]
Comment (mshudrak) [reply .github/workflows/release.yml:22]: Let me think for a bit on this solution vs #100.

[the remark this answers is not in the document]
Comment (HanZ1203) [reply .github/workflows/release.yml:22]: > Let me think for a bit on this solution vs #100.

Sure!

[the remark this answers is not in the document]
Comment (mshudrak) [reply .github/workflows/release.yml:22]: I will try #100 and if we don't like it we will move with this solution.

**Comment by mshudrak** [comment]
We will release manually via Github Actions. It has enough features in order to do that.

**Pull request closed without merging**